pkg/operator/ceph/nfs/spec.go

```diff
@@ -240,7 +240,7 @@ func (r *ReconcileCephNFS) dbusContainer(nfs *cephv1.CephNFS) v1.Container {
 }
 
 func getLabels(n *cephv1.CephNFS, name string) map[string]string {
-	labels := controller.AppLabels(AppName, n.Namespace)
```

Comment:
Do we need `AppLabels()` anymore? What about deleting it, then renaming `PodLabels()` --> `AppLabels()` for more consistent labels everywhere?

Reply:
`AppLabels()` are still used for creating MGR services and some other non-pod things.

Reply:
Right, seems like the more detailed `PodLabels()` could also apply in those cases, if we just rename `PodLabels()` so it's not confusing being applied to non-pods.

Reply:
`PodLabels` aren't applied to k8s resources that don't run pods as far as I can tell.

Reply:
The mgr labels would be helpful at least to be consistent with that daemon type. For the cleanup service, we could set the daemon type to "cleanup". It's not likely to matter for a short-lived daemon like cleanup, so I was thinking simplifying the helpers would be a benefit if we don't need the different helpers.

Reply:
But the problem in my mind is that cleanup isn't a Ceph daemon. It's a Rook job. but it's not something like `mon`, `mgr`, `osd`, `mds`, or `rgw` which are defined by Ceph.

Reply:
I think I am focusing really hard on what the user sees, but I do see desire to make the code cleaner for all us devs. I think having these still be separate is necessary from the user view, but maybe it would be clearer if I renamed `PodLabels` to `CephDaemonAppLabels` or something so it's a little more clear.

Reply:
Ultimately, the original issue was fixed with this change so I don't want to get caught up on the bigger discussion. After the merge conflicts are resolved I can approve.

I guess I'm still wondering why the labels are only for ceph daemons, rather than any daemon. I think of the cleanup job as a rook daemon, so it could have the labels too.

Reply:
Any daemon is identified by `app=<blah>`. Ceph daemons, in my mind, are only daemons produced by the `ceph/ceph` project. So: `mds`, `mgr`, `mon`, `osd`, `rgw`, `nfs`(-ganesha). Admins may need a way to programmatically match between daemons in the context of Ceph CLI and the context of Kubernetes/Rook. I think we can make that a little easier for them if we provide labels that help them tie the Ceph daemon types and IDs together between the two domains.

For my purposes, I'm collecting information from Ceph daemon admin sockets running in a pod and trying to not do any finnicky string of `rook-ceph-<stuff>` to get the daemon type.
https://github.com/BlaineEXE/supportutils-plugin-rook/blob/45589efa624d04eeef6aaae9ead75daeb073a243/rook.sh#L135
https://github.com/BlaineEXE/supportutils-plugin-rook/blob/45589efa624d04eeef6aaae9ead75daeb073a243/rook.sh#L157

Reply:
If the label is `ceph_daemon_type`, then renaming it to CephDeaemonAppLabels() makes sense. I don't see a need to change it to just `daemon_type`.